src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -32,11 +35,20 @@ public void setStep(final EventPublishingStep step) {
         this.response.setStep(step);
     }
 
-    public void setPublishingStatus(final EventPublishingStatus publishingStatus) {
+    // avoid race condition with kafka producer callback
+    synchronized public void setPublishingStatus(final EventPublishingStatus publishingStatus) {
```

Comment:
Yes, it would be simpler that way. Just did it in the last commit https://github.com/zalando/nakadi/pull/208/commits/95472cfec3e8bbf78b1a7d400be4276920cbd06f